NewForce-Cohort-8/tabloidfullstack-the-cage-crushers
# Edit Post Functionality

Description

Edit Post Functionality

Authors need the ability to edit their posts for accuracy and relevance. When viewing a post, they should have an option to edit it. After confirming the edits, the changes should be saved to the database, and authors redirected to the updated post details. If authors choose not to confirm, the post should remain unchanged, and they should return to the list of posts.


Fixes # (issue)

[Trello Ticket 8](https://trello.com/c/y4YrRve2/8-8-edit-a-post)

Type of change

New feature (non-breaking change which adds functionality)

Testing Instructions


1. git checkout main
2. git fetch --a
3. git checkout `ht-editpost`
4. If you don't see the most recent version of my files, git pull origin `ht-editpost`
5. If you haven't created the Tabloid Full Stack database, please do so.
6. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-cage-crushers/blob/main/SQL/01_Tabloid_Create_DB.sql)
7. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-cage-crushers/blob/main/SQL/02_Tabloid_Seed_Data.sql)
8. Click the execute with debugger solid green button to launch the app.
9. Then, run the application in a browser window by running `npm start ` in the client/tabloid folder. You should see the 'Home' page with a navigation bar at the top of the page. If not already logged out, log out (with the link in the navbar), and 'Log in' with the following credentials.
Email: [aotton2@ow.lyx](mailto:aotton2@ow.lyx)
10. Once you're logged in, click on "My Posts" in the nav bar.
11. You will be directed to the "My Post" List where you will see all posts created by the logged in user.
12. You will see the edit button under each created post, click that button to start in-line editing.
13. Change the information for the given post - you have the option to change the title, content, and image URL.
14. Once you have made changes to the post, you will click "save" to successfully send the changes to the database.
15. The page will reload and you will see your changes made to the post.
16. Run` Select * from Post; ` in a SQL query to ensure the changes also reflect in the database. 

Checklist:


-  My code follows the style guidelines of this project
-  I have performed a self-review of my own code
-  I have commented my code, particularly in hard-to-understand areas
-  My changes generate no new warnings or errors
-  I have added test instructions that prove my fix is effective or that my feature works
